webapp/src/main/java/com/epam/esm/controller/GiftCertificateController.java

```diff
@@ -0,0 +1,59 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.data.GiftCertificateDto;
+import com.epam.esm.service.GiftCertificateService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.*;
+
+@RestController
+@RequestMapping("/certificates")
+public class GiftCertificateController {
+
+    private static final String CERTIFICATE_HAS_BEEN_ADDED = "Certificate has been added";
+    private static final String CERTIFICATE_HAS_BEEN_UPDATED = "Certificate has been updated";
+    private static final String CERTIFICATE_HAS_BEEN_DELETED = "Certificate has been deleted";
+    private final GiftCertificateService giftCertificateService;
+
+    @Autowired
+    public GiftCertificateController(GiftCertificateService giftCertificateService) {
+        this.giftCertificateService = giftCertificateService;
+    }
+
+    @GetMapping(value = "/{id}")
+    public GiftCertificateDto getById(@PathVariable int id) {
+        return giftCertificateService.getById(id);
+    }
+
+    @GetMapping
+    public List<GiftCertificateDto> getCertificates(@RequestParam Optional<String> search,
+                                                    @RequestParam Optional<String> tag,
+                                                    @RequestParam Optional<List<String>> sort) {
+        if (!search.isPresent() && !tag.isPresent() && !sort.isPresent()) {
+            return giftCertificateService.getAll();
+        }
+        return giftCertificateService.getWithParameters(search, tag, sort.orElse(Collections.emptyList()));
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
```

Comment:
Please add Location header or use 200 OK status

Reply:
Changed to OK